lib/wallet/http.js

```diff
@@ -1551,6 +1553,103 @@ class HTTP extends Server {
 
       return res.json(200, mtx.getJSON(this.network));
     });
+
+    // SendMany
+    this.post('/wallet/:id/sendmany', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const account = valid.str('account', 'default');
+      const sendTo = valid.array('sendto', []);
+      const minconf = valid.u32('minconf', 1);
+      const subtractFee = valid.bool('subtractfee', false);
+      const passphrase = valid.str('passphrase');
+
+      if (sendTo.length === 0)
+        throw new Error('parameter sendto is required and can not be empty!');
+
+      const cache = req.wallet.hsdTransferResults;
+      const results = [];
+
+      for (const {idempotency_key: idempotencyKey} of sendTo) {
+        if (idempotencyKey && cache.has(idempotencyKey))
+          results.push(cache.get(idempotencyKey));
+      }
+
+      const filteredSendTo = sendTo.filter(element =>
```

Comment:
Just an observation: this could be done in one pass with the above loop, since this is the inverse set.

Not blocking if you prefer to leave it as-is